crates/codspeed/src/fifo.rs

```diff
@@ -0,0 +1,256 @@
+use anyhow::{bail, Context};
+use nix::libc::O_NONBLOCK;
+use nix::sys::stat;
+use nix::unistd::{self, unlink};
+use std::fs::{File, OpenOptions};
+use std::io::Read;
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::{Path, PathBuf};
+
+// !!!!!!!!!!!!!!!!!!!!!!!!
+// !! DO NOT TOUCH BELOW !!
```

Comment:
We could maybe extract this to a `shared` module to make it even more explicit ?